xml/System.Runtime.CompilerServices/Unsafe.xml

```diff
@@ -407,12 +488,33 @@
         <Parameter Name="source" Type="TFrom" RefType="ref" Index="0" FrameworkAlternate="dotnet-plat-ext-2.1;dotnet-plat-ext-2.2;net-5.0;net-6.0;netcore-1.1;netcore-3.0;netcore-3.1" />
       </Parameters>
       <Docs>
-        <typeparam name="TFrom">The type of reference to reinterpret.</typeparam>
-        <typeparam name="TTo">The desired type of the reference.</typeparam>
-        <param name="source">The reference to reinterpret.</param>
-        <summary>Reinterprets the given reference as a reference to a value of type <typeparamref name="TTo" />.</summary>
-        <returns>A reference to a value of type <typeparamref name="TTo" />.</returns>
-        <remarks>To be added.</remarks>
+        <typeparam name="TFrom">The type of managed pointer to reinterpret.</typeparam>
+        <typeparam name="TTo">The desired type of the managed pointer.</typeparam>
+        <param name="source">The managed pointer to reinterpret.</param>
+        <summary>Reinterprets the given managed pointer as a new managed pointer to a value of type <typeparamref name="TTo" />.</summary>
+        <returns>A managed pointer to a value of type <typeparamref name="TTo" />.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This API is conceptually similar to C++'s `reinterpret_cast<>`. It is the caller's responsibility to ensure that the cast is legal. No runtime check will be performed.
+
+Only the managed pointer is reinterpreted. The referenced value itself will remain unchanged. Consider the example below.
+
+```cs
+int[] intArray = new int[] { 0x12345678 }; // a 1-element array
+ref int refToInt32 = ref intArray[0]; // managed pointer to first Int32 in array
+ref short refToInt16 = ref Unsafe.As<int, short>(ref refToInt32); // reinterpret as managed pointer to Int16
+Console.WriteLine(refToInt16);
+```
+
+The output of this program depends on the endianness of the current machine. On big-endian architectures this will output `0x1234`, and on little-endian architectures this will output `0x5678`.
+
+When casting a managed pointer from a narrower type to a wider type, the caller must ensure that dereferencing the pointer will not incur an out-of-bounds access. The caller is also responsible for ensuring that the resulting pointer is properly aligned for the referenced type.
```

Comment:
> properly aligned for the referenced type

Can we link this to where "properly aligned" is specified?

I'm aware (at least I hope so...) on what rules apply here, but a natural question is here: "What is properly aligned for TTo?".

Reply:
This is machine/architecture dependent.

The requirement of ECMA-335 is `byte`/`sbyte` are 1-byte aligned, `short`/`ushort` are 2-byte aligned, and `int`/`uint` are 4-byte aligned.

The alignment of `nint`/`nuint` must be 4-byte aligned on a 32-bit platform and 8-byte aligned on a 64-bit platform.

The alignment of types like `double`, `long`, and `ulong` could be 4-bytes on a 32-bit platform.

The layout and alignment requirements of user-defined types build off the above but can vary from between systems.

Generally its best to just use `Unsafe.Read/WriteUnaligned` if you're unsure and have the JIT optimize it down to a regular read/write on platforms that support it.